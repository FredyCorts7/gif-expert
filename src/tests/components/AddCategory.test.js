import React from 'react';
import { shallow } from 'enzyme';
import AddCategory from '../../components/AddCategory';
import '@testing-library/jest-dom';

describe('Testing to <AddCategory/>', () => {
  const setCategories = jest.fn();
  let wrapper = shallow(<AddCategory setCategories={setCategories} />);

  beforeEach(() => {
    jest.clearAllMocks();
    wrapper = shallow(<AddCategory setCategories={setCategories} />);
  });

  test('should be the same component', () => {
    expect(wrapper).toMatchSnapshot();
  });

  test('should change the textField', () => {
    const valueExpected = 'Hola Kamado';

    const input = wrapper.find('input');
    input.simulate('change', { target: { value: valueExpected } });
    const valueTest = wrapper.find('p').text();

    expect(valueTest).toBe(valueExpected);
  });

  test('should not post information with submit', () => {
    wrapper.find('form').simulate('submit', { preventDefault() {} });

    expect(setCategories).not.toHaveBeenCalled();
  });

  test('should call setCategories and clean textField', () => {
    const expectedValue = 'One Piece';

    wrapper
      .find('input')
      .simulate('change', { target: { value: expectedValue } });
    wrapper.find('form').simulate('submit', { preventDefault() {} });
    const testValue = wrapper.find('input').prop('value');

    expect(setCategories).toBeCalledTimes(1);
    expect(setCategories).toBeCalledWith(expect.any(Function));
    expect(testValue).toBe('');
  });
});
