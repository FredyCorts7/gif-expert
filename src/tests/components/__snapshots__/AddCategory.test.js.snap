// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing to <AddCategory/> should be the same component 1`] = `
<form
  onSubmit={[Function]}
>
  <p />
  <input
    onChange={[Function]}
    type="text"
    value=""
  />
</form>
`;
